share/disambiguation.py

```diff
@@ -142,187 +110,98 @@ def _instance_for_node(self, node):
+            if self.source:
+                qs = model.objects.filter(taxonomy__source=self.source)
 
         if not qs:
             return None
 
-        uri = node.attrs.get('uri')
+        uri = node['uri']
         if uri:
             try:
                 return qs.get(uri=uri)
-            except node.model.DoesNotExist:
+            except model.DoesNotExist:
                 pass
 
-        name = node.attrs.get('name')
+        name = node['name']
         if name:
             try:
                 return qs.get(name=name)
-            except node.model.DoesNotExist:
+            except model.DoesNotExist:
                 pass
         logger.debug('Could not disambiguate subject %s', node)
         return None
 
-    def _query_pair(self, key, value):
-        try:
-            if not value.instance:
+    def _query_pair(self, key, value, instance_map):
+        if isinstance(value, MutableNode):
+            instance = instance_map.get(value)
+            if not instance:
                 return (None, None)
-            return ('{}__id'.format(key), value.instance.id)
-        except AttributeError:
-            return (key, value)
-
-    def _merge_nodes(self, source, replacement):
-        assert source.graph is replacement.graph
-        for k, v in source.attrs.items():
-            if k in replacement.attrs:
-                old_val = replacement.attrs[k]
-                if v == old_val:
-                    continue
-                field = replacement.model._meta.get_field(k)
-                if isinstance(field, DateTimeField):
-                    new_val = max(pendulum.parse(v), pendulum.parse(old_val)).isoformat()
-                else:
-                    # use the longer value, or the first alphabetically if they're the same length
-                    new_val = sorted([v, old_val], key=lambda x: (-len(str(x)), x))[0]
-            else:
-                new_val = source.attrs[k]
-            replacement.attrs[k] = new_val
+            return ('{}__id'.format(key), instance.id)
+        return (key, value)
 
-        from share.models import Person
-        if replacement.model == Person:
-            replacement.attrs['name'] = ''
-            Person.normalize(replacement, replacement.graph)
 
-        source.graph.replace(source, replacement)
+class DisambiguationInfo:
+    def __init__(self, node):
+        self._node = node
+        self.all = self._all()
+        self.any = self._any()
+        self.matching_types = self._matching_types()
 
-    class NodeIndex:
-        def __init__(self):
-            self._index = {}
-            self._info_cache = {}
+    def _all(self):
+        try:
+            all = self._node.model.Disambiguation.all
+        except AttributeError:
+            return ()
+        values = tuple((f, v) for f in all for v in self._field_values(f))
+        assert len(values) == len(all), 'Wrong number of values: {} for "all": {}'.format(values, all)
```

Comment:
`ValueError` would make a bit more sense here, and not mess with pytest.

Reply:
Yeah that makes sense. Kinda knew all those assertions were bad but did them anyway...